lib/generator.go

```diff
@@ -75,58 +83,121 @@ func GenerateCCRs(config *Config, requests chan *request) (err error) {
 	// ccrsPerDay = 1440m / 30m = 40 CCR a day
 	// ccrsTotal = ccrsPerDay * 30d = 1200 per Node
 	if config.DaysBack > 0 {
-		ccrsPerDay = 1440 / config.Interval
-		ccrsTotal = ccrsPerDay * config.DaysBack
+		ccrsPerDay = 1440 / int64(config.Interval)
+		ccrsTotal = ccrsPerDay * int64(config.DaysBack)
 	}
 
 	if config.RunChefClient {
 		chefClient = getAPIClient(config.ClientName, config.ClientKey, config.ChefServerURL)
 	}
 
 	log.WithFields(log.Fields{
-		"nodes":       config.NumNodes,
-		"days_back":   config.DaysBack,
-		"random_data": config.RandomData,
+		"nodes":        config.NumNodes,
+		"days_back":    config.DaysBack,
+		"ccr_per_node": ccrsTotal,
+		"total_ccrs":   ccrsTotal * int64(config.NumNodes),
+		"random_data":  config.RandomData,
+		"goroutines":   config.Threads,
 	}).Info("Generating chef-client runs")
 
 	rand.Seed(time.Now().UTC().UnixNano())
 
+	// Lets try to use a smaller number of goroutines
+	if config.NumNodes > config.Threads {
```

Comment:
This is great to see! We don't need 10,000 go routines for 10,000 threads. :1st_place_medal: